packages/waku/src/lib/plugins/vite-plugin-deploy-cloudflare.ts

```diff
@@ -136,93 +131,36 @@ export function deployCloudflarePlugin(opts: {
-        // exclude strategy
-        const staticPaths: string[] = ['/assets/*'];
-        const paths = getFiles(publicDir);
-        for (const p of paths) {
-          const basePath = path.dirname(p.replace(publicDir, '')) || '/';
-          const name = path.basename(p);
-          const entry =
-            name === 'index.html'
-              ? basePath + (basePath !== '/' ? '/' : '')
-              : path.join(basePath, name.replace(/\.html$/, ''));
-          if (
-            entry.startsWith('/assets/') ||
-            entry.startsWith('/' + WORKER_JS_NAME + '/') ||
-            entry === '/' + WORKER_JS_NAME ||
-            entry === '/' + ROUTES_JSON_NAME ||
-            entry === '/' + HEADERS_NAME
-          ) {
-            continue;
-          }
-          if (!staticPaths.includes(entry)) {
-            staticPaths.push(entry);
-          }
-        }
-        const MAX_CLOUDFLARE_RULES = 100;
-        if (staticPaths.length + 1 > MAX_CLOUDFLARE_RULES) {
-          throw new Error(
-            `The number of static paths exceeds the limit of ${MAX_CLOUDFLARE_RULES}. ` +
-              `You need to create a custom ${ROUTES_JSON_NAME} file in the public folder. ` +
-              `See https://developers.cloudflare.com/pages/functions/routing/#functions-invocation-routes`,
-          );
-        }
-        const staticRoutes: StaticRoutes = {
-          version: 1,
-          include: ['/*'],
-          exclude: staticPaths,
-        };
-        writeFileSync(routesFile, JSON.stringify(staticRoutes));
-      }
-
-      // Move the public files to the root of the dist folder
-      const publicPaths = readdirSync(
-        path.join(outDir, WORKER_JS_NAME, DIST_PUBLIC),
-      );
-      for (const p of publicPaths) {
-        renameSync(
-          path.join(outDir, WORKER_JS_NAME, DIST_PUBLIC, p),
-          path.join(outDir, p),
-        );
-      }
-      rmSync(path.join(outDir, WORKER_JS_NAME, DIST_PUBLIC), {
-        recursive: true,
-        force: true,
-      });
+      const assetsDistDir = path.join(outDir, 'assets');
+      const workerDistDir = path.join(outDir, 'worker');
+      const tempDist = path.join(rootDir, '_dist');
+
+      // Move outDir to tempDist
+      renameSync(outDir, tempDist);
+
+      // Create empty directories
+      mkdirSync(workerDistDir, { recursive: true });
+      mkdirSync(assetsDistDir, { recursive: true });
+
+      // Move tempDist/public to assetsDistDir
+      renameSync(path.join(tempDist, DIST_PUBLIC), assetsDistDir);
+
+      // Move tempDist to workerDistDir
+      renameSync(tempDist, workerDistDir);
+
+      // Traverse assetsDistDir and copy specific files to workerDistDir/public
+      const workerPublicDir = path.join(workerDistDir, DIST_PUBLIC);
+      mkdirSync(workerPublicDir, { recursive: true });
+      copyFiles(assetsDistDir, workerPublicDir, [
+        '.txt',
+        '.html',
+        '.json',
+        '.js',
+        '.css',
+      ]);
```

Comment:
would this break if there were files like a CNAME of favicon?

Reply:
I don't think so. The purpose of this is to copy built public assets that might imported during SSR.

It would be better if we had a manifest of these files to copy rather than making assumptions by file extension.